precompile/testutils/test_predicate.go

```diff
@@ -0,0 +1,87 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package testutils
+
+import (
+	"testing"
+	"time"
+
+	"github.com/ava-labs/subnet-evm/precompile/precompileconfig"
+	"github.com/stretchr/testify/require"
+)
+
+type PredicateTest struct {
+	Config precompileconfig.Config
+
+	ProposerPredicateContext *precompileconfig.ProposerPredicateContext
+
+	StorageSlots         []byte
+	Gas                  uint64
+	GasErr, PredicateErr error
+}
+
+func (test PredicateTest) Run(t testing.TB) {
+	t.Helper()
+	var (
+		gas                  uint64
+		gasErr, predicateErr error
+		predicate            = test.Config.(precompileconfig.ProposerPredicater)
+	)
+
+	gas, gasErr = predicate.PredicateGas(test.StorageSlots)
+	if test.GasErr != nil {
+		require.ErrorIs(t, gasErr, test.GasErr)
+		return // If PredicateGas returns an error, the predicate is invalidated and no further checks are needed
```

Comment:
nit: what does it mean for a predicate to be invalidated?

Reply:
just that it fails. I think I may have wrote invalidated because it fails for a reason other than the verify predicate function returning an error.